project-chip/connectedhomeip
# Implicit admin over PASE is one way

#### Problem
Implicit admin over PASE should only go one way, and not allow
commissionee to perform IM operations willy nilly on the commissioner.
 
 #16531

#### Change overview
- Add accessors to CryptoContext for IsInitiator and IsReponder
- Ensure SecureSession::GetSubjectDescriptor only fills it in for PASE
  if it's the Responder (not the Initiator)
- Add detail to comment in AccessControl::Check about implicit admin
  over PASE (which only occurs on the responder/commissionee, not on
  the initiator/commissioner)

Fixes #16531

#### Testing
- Build and run (chip-all-clusters-app, chip-tool) on Linux x64
- Verify temp logs that subject descriptor is filled correctly for responder
- Verify that if subject descriptor is initiator, access is denied
